cmd/build/main.go

```diff
@@ -19,10 +20,10 @@ import (
 var mgmakeGen []byte
 
 func main() {
-	ctx := mglog.NewContext(context.Background(), "mage-tools-build")
+	ctx := logr.NewContext(context.Background(), mglogr.New("mage-tools-build"))
```

Comment:
`ctx, cancelFunc := signal.NotifyContext(context.Background(), syscall.SIGTERM, syscall.SIGINT)
`

perhaps?

Reply:
I'd rather not trap the signals until we know that we actually handle context cancellation everywhere - haven't started thinking about that yet - I think we need to use https://pkg.go.dev/os/exec#CommandContext everywhere to ensure the top-level program actually exits when the context is cancelled